File: backend/src/models/Message.ts
import mongoose from "mongoose";

const MessageSchema = new mongoose.Schema({
  chatId: {
    type: String,
    required: true,
  },
  userId: {
    type: String,
    required: true,
  },
  text: {
    type: String,
    required: true,
  },
  isUser: {
    type: Boolean,
    required: true,
  },
  image: {
    type: String,
    required: false,
    default: null, // optional image URL or path
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

export default mongoose.model("Message", MessageSchema);

File: lib/models/message.dart
class Message {
  final String id;
  final String chatId;
  final String userId;
  final String text;
  final bool isUser;
  final DateTime createdAt;

  Message({
    required this.id,
    required this.chatId,
    required this.userId,
    required this.text,
    required this.isUser,
    required this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> j) => Message(
    id: j['_id'] as String,
    chatId: j['chatId'] as String,
    userId: j['userId'] as String,
    text: j['text'] as String,
    isUser: j['isUser'] as bool,
    createdAt: DateTime.parse(j['createdAt'] as String),
  );

  Map<String, dynamic> toJson() => {
    'userId': userId,
    'chatId': chatId,
    'text': text,
  };
}
